src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/Subscriptions/DependencySubscriptionsHost.cs

```diff
@@ -141,9 +162,22 @@ public Task OnProjectFactoryCompletedAsync()
             return AddInitialSubscriptionsAsync();
         }
 
+        /// <summary>
+        /// Workaround for CPS bug 375276 which causes double entry on InitializeAsync and exception
+        /// "InvalidOperationException: The value factory has called for the value on the same instance".
+        /// https://dev.azure.com/devdiv/DevDiv/_workitems/edit/375276
+        /// </summary>
+        private async Task EnsureInitialized()
```

Comment:
Its unclear to me if this is still needed @lifengl it looks like  this was not reproduceable on your end?

Reply:
I contemplated removing this, especially after the discussion in #4294, but wasn't confident it was safe to do so. Would be interested to hear @lifengl and @davkean's thoughts about whether this can be removed.